test/SharedFx.UnitTests/SharedFxTests.cs

```diff
@@ -1,14 +1,76 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using System;
+using System.Collections.Generic;
 using System.IO;
+using System.IO.Compression;
+using System.Net;
+using System.Reflection;
 using Newtonsoft.Json.Linq;
 using Xunit;
 
 namespace Microsoft.AspNetCore
 {
     public class SharedFxTests
     {
+
+        [Theory]
+        [MemberData(nameof(GetSharedFxConfig))]
+        public void BaselineTest(SharedFxConfig config)
+        {
+            var previousVersion = TestData.GetPreviousAspNetCoreReleaseVersion();
+            var url = $"https://dotnetcli.blob.core.windows.net/dotnet/aspnetcore/Runtime/" + previousVersion + "/aspnetcore-runtime-internal-" + previousVersion + "-win-x64.zip";
+            var zipName = "assemblies.zip";
+            var nugetAssemblyVersions = new Dictionary<string, Version>();
+
+            var root = TestData.GetDotNetRoot();
+            var dir = Path.Combine(root, "shared", config.Name, config.Version);
+
+            using (var testClient = new WebClient())
```

Comment:
This definitely needs retry logic. Blob storage can be very finicky.

Here's an amazing RetryHelper: https://github.com/aspnet/AspNetCore-Internal/blob/master/Infrastructure/src/TriageBuildFailures/RetryHelpers.cs

And here's an example of how it's used:
https://github.com/aspnet/AspNetCore-Internal/blob/018295b06b95ff24e016cb34151785fd30eb06b7/Infrastructure/src/TriageBuildFailures/GitHub/GitHubClientWrapper.cs#L58